### ISSUE TITLE
Allow OnButtonTyped work with PanelInputType.Game

### ISSUE THREAD

**Issue by dictateurfou** [Facepunch/sbox-issues#3913]
### What it is?

in razor we allow to use OnButtonEvent but he trigger key when release, 
I have an inventory and I want the player to be able to drag and to know at the same time if shift or control is pressed, this during I don't want to take control of the game from them, currently the only solution I have is to return from the game if the person presses shift or control

### What should it be?

just OnButtonTyped work even if PanelInputType is Game  or make another PanelInputType.GameAndUI

**Comment by DevulTj** [comment]
I believe that you want https://wiki.facepunch.com/sbox/pointer-events#pointerpassthroughtogame

**Comment by dictateurfou** [comment]
not exactly I have no problem with the mouse but with the keyboard, in fact when my interface is open I leave control to the user so he can move but also control his inventory the problem is that the only keyboard input that works is when the key is released and I need it when the player presses it.

for the moment I solved the problem by adding a key in game to return to the rootPanel and change
var HashShiftKeyPressed
and HasControlPressed

basically an example of what I had to do
```csharp
	if ( Game.IsClient )
        {
	        var panel = Game.RootPanel as Hud;
	        panel.HasShiftPressed = Input.Down( "Run" );
        }
```

and here is a gif of the expected result (for control quantity he use ctrl or shift when drag)
https://gyazo.com/bb58735154576ad404bdd900196fefa2
